common/national_focus/poland.txt

```diff
@@ -415,17 +415,35 @@ focus_tree = {
 				value = 1
 			}
 			762 = {
-				add_extra_state_shared_building_slots = 2
-				add_building_construction = {
-					type = steel_mill
-					level = 2
+				if = {
+					limit = {
+						free_building_slots = {
+							building = steel_mill
+							size > 1
+						}
+					}
+					add_extra_state_shared_building_slots = 2
+					add_building_construction = {
+						type = steel_mill
+						level = 2
+					}
+				}
+			}
+			if = {
+				limit = {
+					762 = {
+						free_building_slots = {
+							building = steel_mill
+							size < 2
+						}
+					}
+				}
+				add_resource = {
+					type = steel
+					amount = 8 #r56 was 24
+					state = 762
```

Comment:
This should be an if-else @Xaelorichero-02 , this order of operation means there might be sufficient room for the buildings, causing them to be placed, which might in turn reduce the building slot amount enough to also trigger the 2nd reward.